Build: Fix CRA repro generator and e2e test in PnP mode

ndelangen opened the pull request:
One of the CI tasks keeps failing.

Part of the issue was related to CRA:

 - As of CRA 5 `--use-npm` option has been removed.
To force usage of NPM the only hacky way I found is to override `npm_config_user_agent` env var.
See related PR: https://github.com/facebook/create-react-app/pull/11322
And changelog: https://github.com/facebook/create-react-app/releases/tag/v5.0.0

 - CRA 5 was missing some dependencies, causing Yarn to throw in PnP mode. See https://github.com/facebook/create-react-app/pull/11751

And some other things to Vue 3:

> TL;DR: Vue 3 is now the new default version as of Monday, February 7, 2022!
> https://blog.vuejs.org/posts/vue-3-as-the-new-default.html
---


notes:
> 
> This issue suggest clearing npm cache:
> https://github.com/facebook/create-react-app/issues/6776
> 
> Here's another seemingly related issue:
> https://github.com/facebook/create-react-app/issues/1735

gaetanmaisse committed:
Force use of NPM during bootstrap of CRA repros

As of CRA 5 `--use-npm` option has been removed.
To force usage of NPM the only hacky way I found is to override `npm_config_user_agent` env var.

See related PR: https://github.com/facebook/create-react-app/pull/11322
And changelog: https://github.com/facebook/create-react-app/releases/tag/v5.0.0

ndelangen commented:
Locally I'm getting this:

```
🏃 Running cra into /Users/me/Projects/Storybook/storybook-e2e-testing/cra

🏃 Starting for cra latest

{
  name: 'cra',
  version: 'latest',
  appName: 'cra',
  creationPath: '/Users/me/Projects/Storybook/storybook-e2e-testing',
  cwd: '/Users/me/Projects/Storybook/storybook-e2e-testing/cra',
  e2e: true,
  pnp: true,
  framework: 'react',
  generator: 'npx create-react-app@{{version}} {{appName}} --use-npm && cd {{appName}} && echo "FAST_REFRESH=true" > .env && echo "SKIP_PREFLIGHT_CHECK=true" > .env'
}

🏗 Bootstrapping cra project (this might take a few minutes)
npx create-react-app@latest cra --use-npm && cd cra && echo "FAST_REFRESH=true" > .env && echo "SKIP_PREFLIGHT_CHECK=true" > .env
An error occurred while executing: `npx create-react-app@latest cra --use-npm && cd cra && echo "FAST_REFRESH=true" > .env && echo "SKIP_PREFLIGHT_CHECK=true" > .env`
Command output was:

sh: create-react-app@latest: command not found
```

Which is different form the CI.
It would seem the CI is possible running out of resources, based on issues & discussions in the CRA repo.

ndelangen commented:
![Screenshot 2022-01-31 at 17 34 09](https://user-images.githubusercontent.com/3070389/151834329-d2556c01-7456-4d61-baa2-e780c10e4f4f.png)

ndelangen commented:
Seems like the failure happens before it's installed?! And then the install is halted half way in?

![Screenshot 2022-01-31 at 18 03 28](https://user-images.githubusercontent.com/3070389/151839277-829aecc1-9748-4297-9d71-6493fce33f36.png)

ndelangen committed:
fix e2e tests and update the CI version of node

ndelangen commented:
@shilman review away

ndelangen commented:
oof:

```
👷 Building Storybook
yarn build-storybook --quiet
info @storybook/react v6.5.0-alpha.30
info 
info => Cleaning outputDir: /tmp/storybook-e2e-testing/cra/storybook-static
(node:927) DeprecationWarning: --static-dir CLI flag is deprecated, see:

https://github.com/storybookjs/storybook/blob/next/MIGRATION.md#deprecated---static-dir-cli-flag
(Use `node --trace-deprecation ...` to show where the warning was created)
info => Loading presets
info => Copying static files: ./public => ./
ERR! Error: @storybook/core-server tried to access @storybook/manager-webpack5 (a peer dependency) but it isn't provided by its ancestors; this makes the require call ambiguous and unsound.
ERR! 
ERR! Required package: @storybook/manager-webpack5 (via "@storybook/manager-webpack5/prebuilt-manager")
ERR! Required by: @storybook/core-server@virtual:59f05611faa53a966db6158383aad41196d552ccb3146e01020caaa0190b0f10f15edd2bd44449d4b342f2ce9b82f58dd058b81b815efa94943107d0918ecf00#npm:6.5.0-alpha.30 (via /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/)
ERR! 
ERR! Ancestor breaking the chain: @storybook/core@virtual:0c9ab7d7eb0ce0b8929190e024d87075edc0ad1269452bff30850b516430673e7d14d17fd9cf0313acd718ba151b177b7c77ebea1525fe8753ecd5be137cd364#npm:6.5.0-alpha.30
ERR! 
ERR! 
ERR! Require stack:
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-59f05611fa/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.30-762ee8d659-8.zip/node_modules/@storybook/core/dist/cjs/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-59f05611fa/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.30-762ee8d659-8.zip/node_modules/@storybook/core/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-0e7c5eb19e/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.30-aff00ce9ec-8.zip/node_modules/@storybook/react/dist/cjs/server/build.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-0e7c5eb19e/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.30-aff00ce9ec-8.zip/node_modules/@storybook/react/bin/build.js
ERR!     at Function.external_module_.Module._resolveFilename (/tmp/storybook-e2e-testing/cra/.pnp.cjs:35615:55)
ERR!     at Function.external_module_.Module._load (/tmp/storybook-e2e-testing/cra/.pnp.cjs:35414:48)
ERR!     at Module.require (internal/modules/cjs/loader.js:957:19)
ERR!     at require (internal/modules/cjs/helpers.js:88:18)
ERR!     at /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:151:36
ERR!     at async buildStaticStandalone (/tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:150:43)
ERR!     at async buildStatic (/tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:197:5)
ERR!  Error: @storybook/core-server tried to access @storybook/manager-webpack5 (a peer dependency) but it isn't provided by its ancestors; this makes the require call ambiguous and unsound.
ERR! 
ERR! Required package: @storybook/manager-webpack5 (via "@storybook/manager-webpack5/prebuilt-manager")
ERR! Required by: @storybook/core-server@virtual:59f05611faa53a966db6158383aad41196d552ccb3146e01020caaa0190b0f10f15edd2bd44449d4b342f2ce9b82f58dd058b81b815efa94943107d0918ecf00#npm:6.5.0-alpha.30 (via /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/)
ERR! 
ERR! Ancestor breaking the chain: @storybook/core@virtual:0c9ab7d7eb0ce0b8929190e024d87075edc0ad1269452bff30850b516430673e7d14d17fd9cf0313acd718ba151b177b7c77ebea1525fe8753ecd5be137cd364#npm:6.5.0-alpha.30
ERR! 
ERR! 
ERR! Require stack:
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-59f05611fa/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.30-762ee8d659-8.zip/node_modules/@storybook/core/dist/cjs/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-59f05611fa/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.30-762ee8d659-8.zip/node_modules/@storybook/core/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-0e7c5eb19e/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.30-aff00ce9ec-8.zip/node_modules/@storybook/react/dist/cjs/server/build.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-0e7c5eb19e/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.30-aff00ce9ec-8.zip/node_modules/@storybook/react/bin/build.js
ERR!     at Function.external_module_.Module._resolveFilename (/tmp/storybook-e2e-testing/cra/.pnp.cjs:35615:55)
ERR!     at Function.external_module_.Module._load (/tmp/storybook-e2e-testing/cra/.pnp.cjs:35414:48)
ERR!     at Module.require (internal/modules/cjs/loader.js:957:19)
ERR!     at require (internal/modules/cjs/helpers.js:88:18)
ERR!     at /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:151:36
ERR!     at async buildStaticStandalone (/tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:150:43)
ERR!     at async buildStatic (/tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-09bc09ac95/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.30-02f81ce247-8.zip/node_modules/@storybook/core-server/dist/cjs/build-static.js:197:5)
An error occurred while executing: `yarn build-storybook --quiet`
```

ndelangen committed:
improve

ndelangen commented:
I added artifacts to CI to preserve the end state of the e2e dir, and it seems like it too has the dependiencies in place:
 https://circle-production-customer-artifacts.s3.amazonaws.com/picard/56eb8bc6c9e77c0001c7c0e8/61f954b2f7ae9d503f15d599-0-build/artifacts/e2e/cra/package.json?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Date=20220201T155817Z&X-Amz-SignedHeaders=host&X-Amz-Expires=59&X-Amz-Credential=AKIAJR3Q6CR467H7Z55A%2F20220201%2Fus-east-1%2Fs3%2Faws4_request&X-Amz-Signature=1907ed47508487cb5bf7b8a4372520becf5f0a797a9d30dc9dfdbf02805b0685

ndelangen commented:
```
Ancestor breaking the chain: @storybook/core
```

And this in storybook/core:
```
  "peerDependencies": {
    "@storybook/builder-webpack5": "6.5.0-alpha.30",
    "react": "^16.8.0 || ^17.0.0",
    "react-dom": "^16.8.0 || ^17.0.0",
    "webpack": "*"
  },
  "peerDependenciesMeta": {
    "@storybook/builder-webpack5": {
      "optional": true
    },
    "typescript": {
      "optional": true
    }
  },
```

@shilman does core really need webpack5 as a dependency?!?

ndelangen committed:
cleanup

shilman commented:
Probably technically yes but I couldn't tell you why from memory (and am afk right now). Maybe @merceyz remembers.

ndelangen committed:
add missing per deps

ndelangen committed:
Merge branch 'next' into fix/e2e-cra

# Conflicts:
#	lib/core/package.json
#	yarn.lock

ndelangen committed:
might as well try this

ndelangen committed:
set version of @mdx-js/react to 1.x.x

ndelangen commented on lib/cli/src/generators/baseGenerator.ts:
```diff
@@ -87,7 +87,7 @@ export async function baseGenerator(
 
   const yarn2Dependencies =
-    packageManager.type === 'yarn2' ? ['@storybook/addon-docs', '@mdx-js/react'] : [];
+    packageManager.type === 'yarn2' ? ['@storybook/addon-docs', '@mdx-js/react@1.x.x'] : [];
```
@shilman version lock it to v1

ndelangen committed:
sigh

ndelangen commented:
![Screenshot 2022-02-01 at 19 44 09](https://user-images.githubusercontent.com/3070389/152030997-30a98ccf-2d9e-4398-8474-85a0b8c3d080.png)

ndelangen committed:
I hope this is the last one

ndelangen commented:
reminder to remove the saving of artifacts (or ZIP them first)

gaetanmaisse commented:
@ndelangen I started some investigations on that a couple of weeks ago, see https://github.com/storybookjs/storybook/pull/17284

ndelangen commented:
new error:
```
ERR! Module not found: Error: Can't resolve 'assert' in '/root/.yarn/berry/cache/doctrine-npm-3.0.0-c6f1615f04-8.zip/node_modules/doctrine/lib'
ERR! 
ERR! BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
ERR! This is no longer the case. Verify if you need this module and configure a polyfill for it.
ERR! 
ERR! If you want to include a polyfill, you need to:
ERR! 	- add a fallback 'resolve.fallback: { "assert": require.resolve("assert/") }'
ERR! 	- install 'assert'
ERR! If you don't want to include a polyfill, you can use an empty module like this:
ERR! 	resolve.fallback: { "assert": false }
ERR! ModuleNotFoundError: Module not found: Error: Can't resolve 'assert' in '/root/.yarn/berry/cache/doctrine-npm-3.0.0-c6f1615f04-8.zip/node_modules/doctrine/lib'
ERR! 
ERR! BREAKING CHANGE: webpack < 5 used to include polyfills for node.js core modules by default.
ERR! This is no longer the case. Verify if you need this module and configure a polyfill for it.
```

ndelangen commented:
thanks @gaetanmaisse 

I think I'll try it

ndelangen committed:
fix?

ndelangen commented:
Looks like the storybook is now build and hosted, but according to the cypress video gets stuck in a loading state

ndelangen committed:
try and zip the & see if adding 'assert' polyfill helps

ndelangen committed:
what do I say at this point?

ndelangen commented:
So I downloaded the generated result from CI, and can reproduce what's happening.

When I downgrade it to yarn classic then it actually works.

with pnp, it's broken...

I can't figure out why though... 
The build-storybook command just exits (0) half-way through the build process...

ndelangen commented:
OK, I'm going to shelf this work, until @gaetanmaisse has time to work on it together (next week?).

The current state is broken, for unknown reasons. (see above).

I'm going to extract the MSX v2 fix, and open a separate PR, to be released ASAP @shilman

ndelangen committed:
Merge branch 'next' into fix/e2e-cra

ndelangen committed:
well, that's one way to do it I guess, sorry

ndelangen committed:
Merge branch 'next' into fix/e2e-cra

# Conflicts:
#	addons/docs/package.json
#	addons/essentials/package.json
#	lib/cli/src/versions.ts
#	lib/core/package.json
#	yarn.lock

merceyz commented:
> Probably technically yes but I couldn't tell you why from memory (and am afk right now). Maybe @merceyz remembers.

`@storybook/core` has a peer dependency on `@storybook/builder-webpack5` because it needs to provide it to its dependency `@storybook/core-server` which tries to import it here https://github.com/storybookjs/storybook/blob/5d9a1c811f84cd734d93e48eb408bd5a1928cb01/lib/core-server/src/build-static.ts#L123

ndelangen committed:
Merge branch 'next' into fix/e2e-cra

# Conflicts:
#	addons/docs/package.json
#	addons/essentials/package.json
#	lib/cli/src/versions.ts
#	lib/core/package.json
#	yarn.lock

ndelangen committed:
change back to use pnp

ndelangen commented on lib/builder-webpack5/src/preview/iframe-webpack.config.ts:
```diff
@@ -251,6 +251,7 @@ export default async (options: Options & Record<string, any>): Promise<Configura
       },
       fallback: {
         path: require.resolve('path-browserify'),
+        assert: require.resolve('browser-assert'),
```
@gaetanmaisse here's the fix for where you ran into

ndelangen commented on lib/core/package.json:
```diff
@@ -45,6 +45,7 @@
   },
   "peerDependencies": {
     "@storybook/builder-webpack5": "6.5.0-alpha.39",
+    "@storybook/manager-webpack5": "6.5.0-alpha.39",
```
@gaetanmaisse says we can remove this part, because it's specified below, and yarn would be ok with that.

shilman replied:
Can we do this for all optional peer dependencies? If so, I think that could be an improvement

ndelangen committed:
fix a new error

ndelangen committed:
remove peerDependencies that are already specified in peerDependenciesMeta that were only added to keep yarn with pnp happy

ndelangen committed:
add regenerator-runtime to where it is needed

ndelangen committed:
add excemption to lint warnings

ndelangen committed:
fix linting

ndelangen commented:
@gaetanmaisse this line might be a problem:
https://github.com/storybookjs/storybook/blob/b1e0534204cea67794dba7852e04e20f5bc477a3/lib/core-common/src/utils/paths.ts#L25
though it seems this code isn't used at all!

and this one:
https://github.com/storybookjs/storybook/blob/eb2fb1c96d9c58426c430773dd68d041343b6112/lib/core-common/src/utils/es6Transpiler.ts#L6-L38

What does something like this to a --pnp enabled repo?:
https://github.com/storybookjs/storybook/blob/5d9a1c811f84cd734d93e48eb408bd5a1928cb01/lib/builder-webpack4/src/preview/iframe-webpack.config.ts#L238

gaetanmaisse committed:
Add some dependencies in CRA template generator

From the tests I did locally, adding both these deps is fixing Storybook in a CRA app with Yarn PnP.

gaetanmaisse commented:
@ndelangen with the last commit I pushed it unblocks webpack build, I'm not sure to understand why tho'... 
Then we are facing a classic dependency not declared issue, but it's on CRA side: 

```
ERR! Module build failed (from ./.yarn/__virtual__/babel-loader-virtual-53dbd4e05d/4/root/.yarn/berry/cache/babel-loader-npm-8.2.3-855681b984-8.zip/node_modules/babel-loader/lib/index.js):
ERR! Error: [BABEL] /tmp/storybook-e2e-testing/cra/src/stories/Button.stories.jsx: babel-preset-react-app tried to access @babel/plugin-proposal-private-property-in-object, but it isn't declared in its dependencies; this makes the require call ambiguous and unsound.
ERR! 
ERR! Required package: @babel/plugin-proposal-private-property-in-object
ERR! Required by: babel-preset-react-app@npm:10.0.1 (via /root/.yarn/berry/cache/babel-preset-react-app-npm-10.0.1-a9a19b630a-8.zip/node_modules/babel-preset-react-app/)
ERR! 
ERR! Require stack:
ERR! - /root/.yarn/berry/cache/babel-preset-react-app-npm-10.0.1-a9a19b630a-8.zip/node_modules/babel-preset-react-app/create.js
ERR! - /root/.yarn/berry/cache/babel-preset-react-app-npm-10.0.1-a9a19b630a-8.zip/node_modules/babel-preset-react-app/index.js
ERR! - /root/.yarn/berry/cache/@babel-core-npm-7.17.2-bc12e43b41-8.zip/node_modules/@babel/core/lib/config/files/module-types.js
ERR! - /root/.yarn/berry/cache/@babel-core-npm-7.17.2-bc12e43b41-8.zip/node_modules/@babel/core/lib/config/files/configuration.js
ERR! - /root/.yarn/berry/cache/@babel-core-npm-7.17.2-bc12e43b41-8.zip/node_modules/@babel/core/lib/config/files/index.js
ERR! - /root/.yarn/berry/cache/@babel-core-npm-7.17.2-bc12e43b41-8.zip/node_modules/@babel/core/lib/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@rollup-plugin-babel-virtual-a4dbb1aebf/4/root/.yarn/berry/cache/@rollup-plugin-babel-npm-5.3.0-944843ffd9-8.zip/node_modules/@rollup/plugin-babel/dist/index.js
ERR! - /root/.yarn/berry/cache/workbox-build-npm-6.4.2-83cd657f6a-8.zip/node_modules/workbox-build/build/lib/bundle.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/workbox-webpack-plugin-virtual-8ad76104b5/4/root/.yarn/berry/cache/workbox-webpack-plugin-npm-6.4.2-249a27d315-8.zip/node_modules/workbox-webpack-plugin/build/generate-sw.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/workbox-webpack-plugin-virtual-8ad76104b5/4/root/.yarn/berry/cache/workbox-webpack-plugin-npm-6.4.2-249a27d315-8.zip/node_modules/workbox-webpack-plugin/build/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/react-scripts-virtual-4edbc0c6ff/4/root/.yarn/berry/cache/react-scripts-npm-5.0.0-14ea4c40dd-8.zip/node_modules/react-scripts/config/webpack.config.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-preset-create-react-app-virtual-5ea6eeeb69/4/root/.yarn/berry/cache/@storybook-preset-create-react-app-npm-4.0.1-a0a290772c-8.zip/node_modules/@storybook/preset-create-react-app/dist/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-common-virtual-87a23aa106/4/root/.yarn/berry/cache/@storybook-core-common-npm-6.5.0-alpha.39-70fc45a81e-8.zip/node_modules/@storybook/core-common/dist/cjs/presets.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-common-virtual-87a23aa106/4/root/.yarn/berry/cache/@storybook-core-common-npm-6.5.0-alpha.39-70fc45a81e-8.zip/node_modules/@storybook/core-common/dist/cjs/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-server-virtual-8678751edd/4/root/.yarn/berry/cache/@storybook-core-server-npm-6.5.0-alpha.39-7ed1ca228d-8.zip/node_modules/@storybook/core-server/dist/cjs/index.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-eb80802c10/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.39-fe23192354-8.zip/node_modules/@storybook/core/dist/cjs/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-core-virtual-eb80802c10/4/root/.yarn/berry/cache/@storybook-core-npm-6.5.0-alpha.39-fe23192354-8.zip/node_modules/@storybook/core/server.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-9b12b5bc45/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.39-48dab45a83-8.zip/node_modules/@storybook/react/dist/cjs/server/build.js
ERR! - /tmp/storybook-e2e-testing/cra/.yarn/__virtual__/@storybook-react-virtual-9b12b5bc45/4/root/.yarn/berry/cache/@storybook-react-npm-6.5.0-alpha.39-48dab45a83-8.zip/node_modules/@storybook/react/bin/build.js (While processing: "/root/.yarn/berry/cache/babel-preset-react-app-npm-10.0.1-a9a19b630a-8.zip/node_modules/babel-preset-react-app/index.js")
ERR!     at Function.external_module_.Module._resolveFilename (/tmp/storybook-e2e-testing/cra/.pnp.cjs:35068:55)
```

As a workaround to see what's the status without this issue, I copied the project locally and added the following in the `.yarnrc.yml`:
```
packageExtensions:
  babel-preset-react-app@*:
    dependencies:
      "@babel/plugin-proposal-private-property-in-object": "*"
```

(I also linked all the Storybook packages to have all the fixed done in this PR: `yarn link --all [LINK_TO_YOUR_CLONE_OF_THE_MONOREPO]`).

I would be glad to discuss this with you to define our next move!

ndelangen commented:
@gaetanmaisse that sounds like CRA itself has a problem with pnp then?

merceyz commented:
I've fixed that issue in https://github.com/facebook/create-react-app/pull/11751 but it hasn't been merged yet. The fix is included in the canary releases of Yarn though if you want to use that.
https://github.com/yarnpkg/berry/blob/3cfb8bd988426e1bd062a085e8512c4c12323846/packages/plugin-compat/sources/extensions.ts#L740-L745

ndelangen commented:
@shilman @gaetanmaisse merge and wait for CRA to fix upstream?

shilman commented on lib/cli/src/repro-generators/scripts.ts:
```diff
@@ -105,6 +107,8 @@ const configureYarn2ForE2E = async ({ cwd }: Options) => {
 const generate = async ({ cwd, name, appName, version, generator }: Options) => {
   const command = generator.replace(/{{appName}}/g, appName).replace(/{{version}}/g, version);
 
+  console.log('generate', { command, generator });
```
```suggestion
```

shilman commented on app/react/package.json:
```diff
@@ -86,6 +86,18 @@
     "@babel/core": {
       "optional": true
     },
+    "@storybook/builder-webpack4": {
```
Why do we need to add these?

gaetanmaisse replied:
Because there is `@storybook/core` as dep and it has `@storybook/builder-webpack4` as optional peer dep itself, so we need to add it to satisfy transitivity. Maël did a post about it some time ago: https://dev.to/arcanis/implicit-transitive-peer-dependencies-ed0

shilman commented on addons/docs/package.json:
```diff
@@ -140,29 +140,25 @@
     "@storybook/angular": {
       "optional": true
     },
+    "@storybook/builder-webpack4": {
```
Why do we need to add these?

shilman reviewed (commented):
Generally looks great! Some questions for discussion below

gaetanmaisse commented:
@ndelangen I will review the comments and do the needed changes later today ;)

gaetanmaisse committed:
Add tmp workaround to fix CRA repro in PnP mode

gaetanmaisse committed:
Merge branch 'next' into fix/e2e-cra

gaetanmaisse committed:
Rely on `peerDependenciesMeta` only for some packages

gaetanmaisse committed:
Add missing peer dep in addon-essentials

gaetanmaisse committed:
Ignore eslint warning

We will need to discuss the `import/no-extraneous-dependencies` rule config

gaetanmaisse committed:
Use fixed versions for Vue packages in repro generator

> TL;DR: Vue 3 is now the new default version as of Monday, February 7, 2022!
> https://blog.vuejs.org/posts/vue-3-as-the-new-default.html

gaetanmaisse committed:
Remove wrongly committed packageExtensions

gaetanmaisse committed:
Update yarn.lock
